autogen/oai/client.py

```diff
@@ -287,22 +287,42 @@ def yes_or_no_filter(context, response):
 
     def _completions_create(self, client, params):
         completions = client.chat.completions if "messages" in params else client.completions
-        # If streaming is enabled, has messages, and does not have functions, then
-        # iterate over the chunks of the response
-        if params.get("stream", False) and "messages" in params and "functions" not in params:
+        # If streaming is enabled and has messages, then iterate over the chunks of the response.
+        if params.get("stream", False) and "messages" in params:
             response_contents = [""] * params.get("n", 1)
             finish_reasons = [""] * params.get("n", 1)
             completion_tokens = 0
 
             # Set the terminal text color to green
             print("\033[32m", end="")
 
+            # Prepare for potential function call
+            full_function_call = None
             # Send the chat completion request to OpenAI's API and process the response in chunks
             for chunk in completions.create(**params):
                 if chunk.choices:
                     for choice in chunk.choices:
                         content = choice.delta.content
+                        function_call_chunk = choice.delta.function_call
                         finish_reasons[choice.index] = choice.finish_reason
+
+                        # Handle function call
+                        if function_call_chunk:
+                            if hasattr(function_call_chunk, "name") and function_call_chunk.name:
+                                if full_function_call is None:
+                                    full_function_call = {"name": "", "arguments": ""}
+                                full_function_call["name"] += function_call_chunk.name
+                                completion_tokens += 1
```

Comment:
Is this only one token? I suggest use `count_token`

Reply:
You're probably correct, and `completion_token += 1` should probably not be used anywhere. I was just following the current convention. Let's move that to a new PR since it's not something this PR introduced.

I'll do it right after reorganizing my local. #1164